recipe/patch_yaml/arm-variant.yaml

```diff
@@ -0,0 +1,59 @@
+# https://github.com/conda-forge/arm-variant-feedstock/issues/6 arm-variant should be a run
+# dependency not a run constrained. We want to move the dependency from run_constrained to
+# run, but the patcher raises an error if we do that, so we just add the dependency to run.
+if:
+  subdir_in:
+    - linux-aarch64
+    - noarch
+  has_constrains: "arm-variant * sbsa"
+  timestamp_lt: 1744158002000
+then:
+  - add_depends: "arm-variant * sbsa"
+  # - remove_constrains: "arm-variant * sbsa"
+---
+if:
+  subdir_in:
+    - linux-aarch64
+    - noarch
+  has_constrains: "arm-variant * tegra"
+  timestamp_lt: 1744158002000
+then:
+  - add_depends: "arm-variant * tegra"
+  # - remove_constrains: "arm-variant * tegra"
+---
+# https://github.com/conda-forge/cuda-nvcc-impl-feedstock/issues/42 an arm-variant
+# dependency is needed to be added to the outputs of cuda-nvcc-impl-feedstock because
+# run_exports do not work for noarch packages
+if:
+  timestamp_lt: 1744158002000
+  subdir_in:
+    - linux-aarch64
+    - noarch
+  artifact_in:
+    - cuda-crt-dev_linux-aarch64-12.8.93-h579c4fd_2.conda
+    - cuda-crt-tools-12.8.93-h579c4fd_2.conda
+    - cuda-nvcc-dev_linux-aarch64-12.8.93-h4310d6a_2.conda
+    - cuda-nvcc-impl-12.8.93-h614329b_2.conda
+    - cuda-nvcc-tools-12.8.93-h614329b_2.conda
+    - cuda-nvvm-dev_linux-aarch64-12.8.93-h579c4fd_2.conda
+    - cuda-nvvm-impl-12.8.93-h614329b_2.conda
+    - cuda-nvvm-tools-12.8.93-h614329b_2.conda
+then:
+  - add_depends: "arm-variant * sbsa"
```

Comment:
Are there any others that are missing `arm-variant`, which should have it?

Reply:
I dunno what you mean. Do you want to patch every package that was built for aarch64 before the arm-variant package existed?

These are the package which have been built since the tegra arm variant was added to the feedstock.

Reply:
Right are there past SBSA packages that are not being constrained as such that we should be constraining?

The answer very well may be no. Though it is worth verifying that is true before adding Tegra as an option

Reply:
Right are there past SBSA packages that are not being constrained as such that we should be constraining?

The answer very well may be no. Though it is worth verifying that is true before adding Tegra as an option

Reply:
Do you want to patch every NVIDIA package that was built for aarch64 before the arm-variant package existed?

We could do that. It would be a large patch, but it would be the safest for theoretical tegra users because they would not accidentally be able to install sbsa packages if they are using the arm-variant package correctly.

Reply:
Think we can limit this to CUDA 12

We can ignore CUDA 11 as we don't plan to add Tegra support there

Let's assess how many packages that would be. It may be large or it may not. We should be able to get a list though

Reply:
There was no easy way that I could think of which would filter the packages to only NVIDIA packages. The next best thing I could think of was to filter as follows:

```yaml
  subdir_in:
    - linux-aarch64
    - noarch
  has_depends: "cuda-version?( *)"
  not_has_depends: "arm-variant?( *)"
  not_has_constrains: "arm-variant?( *)"
```
Since the `cuda-version` package is exported by CTK12+ compiler package, only CTK12 packages will be affected. Also only packages which do not already have an arm-variant constraint are affected.

Reply:
Thanks Daniel! 🙏

Think this is a reasonable first pass. Have tried to refine this to CUDA Toolkit packages in the suggestions below. We may want to refine it further

This way other packages can opt-in to this patch as needed and we can better track which are affected